src/libraries/Native/Unix/System.Native/pal_signal.c

```diff
@@ -309,21 +366,68 @@ int32_t InitializeSignalHandlingCore()
     }
 
     // Finally, register our signal handlers
-    // We don't handle ignored SIGINT/SIGQUIT signals. If we'd setup a handler, our child
-    // processes would reset to the default on exec causing them to terminate on these signals.
+    // We don't handle ignored SIGINT/SIGQUIT/SIGHUP/SIGTERM signals. If we'd
+    // setup a handler, our child processes would reset to the default on exec
+    // causing them to terminate on these signals.
     InstallSignalHandler(SIGINT , /* skipWhenSigIgn */ true);
     InstallSignalHandler(SIGQUIT, /* skipWhenSigIgn */ true);
+    InstallSignalHandler(SIGHUP, /* skipWhenSigIgn */ true);
+    InstallSignalHandler(SIGTERM, /* skipWhenSigIgn */ true);
+
     InstallSignalHandler(SIGCONT, /* skipWhenSigIgn */ false);
     InstallSignalHandler(SIGCHLD, /* skipWhenSigIgn */ false);
     InstallSignalHandler(SIGWINCH, /* skipWhenSigIgn */ false);
 
     return 1;
 }
 
+int32_t SystemNative_RegisterForPosixSignal(PosixSignal signal, PosixSignalHandler signalHandler)
+{
+    assert(signalHandler != NULL);
+    assert(g_posixSignalHandler == NULL || g_posixSignalHandler == signalHandler);
+
+    if (signal >= 0 || -signal > PosixSignalCount)
+    {
+        errno = EINVAL;
+        return 0;
+    }
+
+    g_posixSignalHandler = signalHandler;
+    g_signalHasRegistrations[-signal] = true;
+
+    return 1;
+}
+
+void SystemNative_UnregisterForPosixSignal(PosixSignal signal)
+{
+    assert(signal < 0 && -signal <= PosixSignalCount);
+
+    g_signalHasRegistrations[-signal] = false;
+}
+
+void SystemNative_HandlePosixSignal(PosixSignal signal)
```

Comment:
I find this name confusing. It calls the previously registered handlers and it only does it for termination signals. The name seems to suggest that the native PAL is going to handle it.

Reply:
This method is called to perform the default (non-cancelled) behavior the runtime has for the `PosixSignal`.

What may be a better name?

Reply:
I'll rename it to `SystemNative_DefaultSignalHandler`.

Reply:
After struggling with attempts to find a better name, I think that the following might be a usable one:
```
SystemNative_InvokePreviousPosixSignalHandler
```
Little long, but doesn't seem that bad. What do you think?

Reply:
Based on the implementation, I've now named it `HandleNonCanceledPosixSignal`.